### DISCUSSION TITLE
Why can't I disable Github Pages?

### DISCUSSION THREAD

**Discussion by lucypero** [community/community#21771] [Other Feature Feedback, Questions, & Ideas]
<p>I understand you can by deleting the gh-pages branch, but why is that needed? There is a button to disable gh-pages, but it’s greyed out. It’s confusing.</p>
<p><img src="https://user-images.githubusercontent.com/11927498/181066815-e1fdec5a-4af7-4793-b829-aa591f923728.png" alt="xgAZYjr" data-base62-sha1="kFpZORQcXdxKT14jeL1aXR6PBai" width="341" height="318"></p>

**Comment by daviddarnes** [comment, marked as answer]
<p>I think it’s to account for legacy behaviour. Before this option was avialable you could just use a gh-pages branch to trigger GitHub Pages to render. Anyone who creates a branch locally with the same name and pushes it up should get a Pages site automatically. Best thing to do is to branch gh-pages to a name other than this and then turn it off and delete the old branch.</p>

**Comment by andy5995** [comment]
<p>There is a button to disable gh-pages, but it’s greyed out. It’s confusing</p>
<blockquote>
<p><a class="mention" href="/matiaspi">@matiaspi</a> wrote:</p>
</blockquote>
<blockquote>
<p>I understand you can by deleting the gh-pages branch, but why is that needed? There is a button to disable gh-pages, but it’s greyed out. It’s confusing.</p>
</blockquote>
<p>I agree.</p>

**Comment by srfrnk** [comment]
<p>What if you fork a repo with GitHub Pages and you need to keep the branch to be able to create Pull Requests?</p>
<p>Every time I push to my fork it sends me an EMail saying the custom domain is taken.</p>
<p>I tryed changing the domain for my fork but that didn’t help. I can’t disable it as well.</p>
<p>So basically I’m stuck.</p>
<p>Any ideas?</p>

**Comment by dhimmel** [comment]
<p>Thanks for the explanation of how it’s related to legacy behavior. Not being able to disable GitHub Pages when we have a <code>gh-pages</code> branch has <a href="https://github.com/greenelab/meta-review/issues/138#issuecomment-480500119" rel="nofollow noopener">become an annoyance</a> for our project. Frankly, I don’t understand why the “Disable GitHub Pages” options needs to be innaccessible. As long as it is opt-in, I don’t see how it interferes with any legacy behavior or repositories. </p>

**Comment by KEINOS** [comment]
<blockquote>
<p><a class="mention" href="/srfrnk">@srfrnk</a> wrote:</p>
</blockquote>
<blockquote>
<p>What if you fork a repo with GitHub Pages and you need to keep the branch to be able to create Pull Requests?</p>
</blockquote>
<blockquote>
<p>Every time I push to my fork it sends me an EMail saying the custom domain is taken.</p>
</blockquote>
<p><strong>I agree.</strong> As of 2019/06/12, exactly the same thing is happening to me.</p>
<p><a href="https://help.github.com/en/articles/unpublishing-a-project-pages-site" rel="nofollow noopener">GitHub</a><a href="https://help.github.com/en/articles/unpublishing-a-project-pages-site" rel="nofollow noopener">Help</a> didn’t help either. Because  <strong>deleting the “gh-pages” branch is not an option</strong> since I’d like to contribute translate their web pages.</p>
<p>My current workaround is to switch the GitHub Pages’ source to “<em>master</em>” branch, which is also grayed out but choosable. Since this branch doesn’t contain the “<em>CNAME</em>” file.</p>
<p>But still, this exposes the forked master to <em><a href="http://myname.github.io" rel="nofollow noopener">myname.github.io</a></em> and I feel uncomfortable. </p>
<p>What I want is just to choose “None” to disable the GitHub Pages. :sob:</p>
<ul>
<li>Browser: Chrome 74.0.3729.169 + macOS: Mojave (OSX 10.14.5)</li>
</ul>
